Add documentation on remote recovery (#39483)
elastic/elasticsearch: docs/reference/ccr/remote-recovery.asciidoc

[role="xpack"]
[testenv="platinum"]
[[remote-recovery]]
== Remote recovery

> 

Reviewer: I think you need a sentence or two of context at the beginning of this page. For example:

```suggestion
When you create a follower index, you cannot use it until it is fully initialized. 
```

> Remote recovery is the process used to build a new copy of a shard on a follower

Reviewer: ```suggestion
 The _remote recovery_ process builds a new copy of a shard on a follower
```

node by copying data from the primary shard in the leader cluster. {es} uses this
remote recovery process to bootstrap a follower index using the data from the

> leader index. This allows the follower to receive a copy of the current state of

Reviewer: ```suggestion
leader index. This process provides the follower with a copy of the current state of
```

the leader index, even if a complete history of changes is not available on the
leader due to Lucene segment merging.

Remote recovery is a network intensive process that transfers all of the Lucene
segment files from the leader cluster to the follower cluster. The follower
requests that a recovery session be initiated on the primary shard in the leader
cluster. The follower then requests file chunks concurrently from the leader. By

> default, the the process concurrently requests `5` large `1mb` file chunks as remote

Reviewer: ```suggestion
default, the process concurrently requests `5` large `1mb` file chunks as remote
```

> recovery is designed to support leader and follower clusters with high network

Reviewer: ```suggestion
behavior is designed to support leader and follower clusters with high network
```

latency between them.

> Information about an in-progress remote recovery can be obtained using the

Reviewer: ```suggestion
You can obtain information about an in-progress remote recovery by using the
```

{ref}/cat-recovery.html[recovery API] on the follower cluster. Remote recoveries are implemented
using the {ref}/modules-snapshots.html[snapshot and restore] infrastructure. This means that

> on-going remote recoveries will be labelled as type `snapshot` in the recovery api.

Reviewer: ```suggestion
on-going remote recoveries are labelled as type `snapshot` in the recovery API.
```

Reviewer: Is there some way to know when the recovery process is complete (other than using the recovery API)?

Reviewer 2: The recovery API is the primary. You could probably use the `cat indices` api to check if the index is green. You can also add a parameter to a follower request to wait until the process is completed. However, that is documented on the put follow request page.

> The following setting can be used to rate-limit the data transmitted during remote

Reviewer: We generally store these types of settings in the Elasticsearch Reference in pages like the ones linked here: https://www.elastic.co/guide/en/elasticsearch/reference/master/settings-xpack.html

I am happy to create that page if you agree.

Reviewer 2: Yeah that makes sense.

Reviewer: I've added those changes.  You can preview the new page in the Elasticsearch Reference here:
https://doc-review-63bf6.firebaseapp.com/es2/ccr-settings.html 

I've also updated the Remove recovery page in the Stack Overview to link to that content, which you can preview here:
https://doc-review-63bf6.firebaseapp.com/x4/remote-recovery.html
